clients/rospy/src/rospy/impl/tcpros_service.py

```diff
@@ -137,6 +136,7 @@ def contact_service(resolved_name, timeout=10.0):
                 if first:
                     first = False
                     rospy.core.logerr("wait_for_service(%s): failed to contact [%s], will keep trying"%(resolved_name, uri))
+            time.sleep(0.3)
```

Comment:
With the sleep being moved here the call is not retried immediately but during the 0.3s sleep the call is not being retried yet. That seems to be not the idea of the code. Which exception is being reused in your case and what timeout are you using?